OmsAgent/Fairfax/omsagent_ff.py

```diff
@@ -430,6 +428,196 @@ def check_workspace_id_and_key(workspace_id, workspace_key):
+        if ('illegal option' not in output.lower()
+                and 'unknown option' not in output.lower()):
+            if exit_code is 0:
+                hutil_log_info('According to {0}, SCOM port is ' \
+                               'open'.format(omsadmin_cmd))
+                scom_port_open = True
+            elif exit_code is 1:
+                hutil_log_info('According to {0}, SCOM port is not ' \
+                               'open'.format(omsadmin_cmd))
+                return
+
+    # If omsadmin.sh option is not available, use omiconfigeditor
+    if (scom_port_open is None and os.path.exists(OMIConfigEditorPath)
+            and os.path.exists(OMIServerConfPath)):
+        omi_cmd = '{0} httpsport -q 1270 < {1}'.format(OMIConfigEditorPath,
+                                                       OMIServerConfPath)
+        exit_code, output = run_get_output(omi_cmd, False, False)
+        # Guard against older omiconfigeditor versions
+        if ('illegal option' not in output.lower()
+                and 'unknown option' not in output.lower()):
+            if exit_code is 0:
+                hutil_log_info('According to {0}, SCOM port is ' \
+                               'open'.format(omi_cmd))
+                scom_port_open = True
+            elif exit_code is 1:
+                hutil_log_info('According to {0}, SCOM port is not ' \
+                               'open'.format(omi_cmd))
+                return
+
+    # If omiconfigeditor option is not available, directly parse omiserver.conf
+    if scom_port_open is None and os.path.exists(OMIServerConfPath):
+        with open(OMIServerConfPath, 'r') as omiserver_file:
+            omiserver_txt = omiserver_file.read()
+
+        httpsport_search = r'^[\s]*httpsport[\s]*=(.*)$'
+        httpsport_re = re.compile(httpsport_search, re.M)
+        httpsport_matches = httpsport_re.search(omiserver_txt)
+        if (httpsport_matches is not None and
+                httpsport_matches.group(1) is not None):
+            ports = httpsport_matches.group(1)
+            ports = ports.replace(',', ' ')
+            ports_list = ports.split(' ')
+            if '1270' in ports_list:
+                hutil_log_info('SCOM port is listed in ' \
+                               '{0}'.format(OMIServerConfPath))
+                scom_port_open = True
+            else:
+                hutil_log_info('SCOM port is not listed in ' \
+                               '{0}'.format(OMIServerConfPath))
+                return
+        else:
+            hutil_log_info('SCOM port is not listed in ' \
+                               '{0}'.format(OMIServerConfPath))
+            return
+    else:
+        hutil_log_info('SCOM port could not be determined to be open')
+        return
+
+    # Parse the certificate to determine if SCOM issued it
+    if os.path.exists(SCOMCertPath):
+        cert_cmd = 'openssl x509 -in {0} -noout -text'.format(SCOMCertPath)
```

Comment:
Check if openssl is installed and have a proper shutdown when the client tool is not found.

Reply:
Done